openprescribing/matrixstore/management/commands/matrixstore_build.py

```diff
@@ -0,0 +1,31 @@
+"""
+Runs the complete process to build a SQLite file with prescribing data in
+MatrixStore format
+"""
+import os
+
+from django.core.management import BaseCommand
+
+from .matrixstore_init import matrixstore_init
+from .matrixstore_download_practice_stats import matrixstore_download_practice_stats
+from .matrixstore_import_practice_stats import matrixstore_import_practice_stats
+from .matrixstore_download_prescribing import matrixstore_download_prescribing
+from .matrixstore_import_prescribing import matrixstore_import_prescribing
+from .matrixstore_update_bnf_map import matrixstore_update_bnf_map
+
+
+class Command(BaseCommand):
+    help = __doc__
+
+    def add_arguments(self, parser):
+        parser.add_argument('end_date', help='YYYY-MM format')
+        parser.add_argument('sqlite_path')
+
+    def handle(self, end_date, sqlite_path, **kwargs):
+        if not os.path.exists(sqlite_path):
```

Comment:
Why would the path already exist?  Is this about recovering from failures?  And are all the subsequent commands idempotent?

Reply:
Yes, it's about recovering from failures (or deliberate interrupts, which happen a lot during development). Every step is idempotent to the best of my ability.